lib/sync.js

```diff
@@ -68,6 +68,8 @@ module.exports = function (x, options) {
                 if (pkg.main) {
                     if (pkg.main === '.' || pkg.main === './') {
                         pkg.main = 'index';
+                    } else {
+                        pkg.main = pkg.main.match(/(.+?)\.[^.]*$|$/)[1]
```

Comment:
Can you provide a test that fails without your change? It's not clear to me what this is attempting to address.